### PR TITLE
Add support for multiple organizers

### PR DESCRIPTION
**Pull request by vicskf**
Updated the get_recipient function to fetch emails from all organizers linked to the post

Ticket: EXT-212

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -35,6 +35,10 @@ We're always interested in your feedback and our [Help Desk](https://support.the
 
 == Changelog ==
 
+= [TBD] TBD =
+
+* Feature - Add support for events with multiple organizers
```
Comment (cliffordp) [inline README.txt]:
```suggestion
* Feature - Add support for events with multiple organizers. [EXT-NUMBER]
```

File: tribe-ext-organizer-notifications.php [hunk 2]
```diff
@@ -76,18 +76,33 @@ public function generate_email( $other = null, $event_id = null ) {
    }
 
     /**
-     * Get organizer's email address.
+     * Get all organizer's email addresses
```
Comment (cliffordp) [inline tribe-ext-organizer-notifications.php]:
```suggestion
     * Get all organizers' email addresses.
```

File: tribe-ext-organizer-notifications.php [hunk 3]
```diff
@@ -76,18 +76,33 @@ public function generate_email( $other = null, $event_id = null ) {
...
      * @param $post_id
-     * @return string
+     * @return array
      */
     private function get_recipient( $post_id ) {
 
-        // Get the organizer's email
-        $to = html_entity_decode( tribe_get_organizer_email( $post_id ), ENT_COMPAT, 'UTF-8' );
+        //get all organizers asociated to the post
+        $organizers_ids = tribe_get_organizer_ids( $post_id );
```
Comment (cliffordp) [inline tribe-ext-organizer-notifications.php]:
```suggestion
        // Get all organizers associated to the post.
```

And maybe ~$organizers_ids~ to $organizer_ids

File: tribe-ext-organizer-notifications.php [hunk 4]
```diff
@@ -76,18 +76,33 @@ public function generate_email( $other = null, $event_id = null ) {
...
+
+        $to = array();
+
+        //get the email for each organizer
```
Comment (cliffordp) [inline tribe-ext-organizer-notifications.php]:
```suggestion
        // Get the email for each organizer.
```

File: tribe-ext-organizer-notifications.php [hunk 5]
```diff
@@ -76,18 +76,33 @@ public function generate_email( $other = null, $event_id = null ) {
...
+        foreach ( $organizers_ids as $organizer_id ) {
+            $organizer_email = tribe_get_organizer_email( $organizer_id, false );
+
+            //make sure it's a valid email
```
Comment (cliffordp) [inline tribe-ext-organizer-notifications.php]:
```suggestion
            // Make sure it's a valid email.
```

File: tribe-ext-organizer-notifications.php [hunk 6]
```diff
@@ -76,18 +76,33 @@ public function generate_email( $other = null, $event_id = null ) {
...
+            if ( is_email( $organizer_email ) ) {
+                $to[] = $organizer_email;
+            }
+        }
 
-        // Return valid email, or empty string
-        return ( is_email( $to ) ) ? $to : '';
+        if ( empty( $to ) ) {
+            return '';
```
Comment (cliffordp) [inline tribe-ext-organizer-notifications.php]: Match `@return` type
```suggestion
            return [];
```

**Review by cliffordp — CHANGES_REQUESTED** [review]

File: tribe-ext-organizer-notifications.php [hunk 7]
```diff
@@ -1,131 +1,144 @@
...
+		 * @param $other
+		 * @param $event_id
+		 */
+		public function generate_email( $other = null, $event_id = null ) {
+
+			// Get the organizer email address.
+			$to = $this->get_recipient( $event_id );
+
+			// Bail if there's not a valid email for the organizer.
+			if ( '' === $to ) {
```
Comment (vicskf) [inline tribe-ext-organizer-notifications.php:59]: @cliffordp do we need to change how we check $to now that get_recipient() always returns an array?

Comment (cliffordp) [reply tribe-ext-organizer-notifications.php:59]: @vicskf hey look what old comment I found :) Sorry about that - yes, looks like it. Ping me when ready for a fresh review

**Comment by bobr666** [comment]
> Updated the get_recipient function to fetch emails from all organizers linked to the post
> 
> Ticket: EXT-212

I have installed this plug in to Event Tickets Pro, but it is still only emailing the first organiser

File: tribe-ext-organizer-notifications.php [hunk 8]
```diff
@@ -1,131 +1,144 @@
...
+		 *
+		 * @return string
+		 */
+		private function get_subject( $post_id ) {
+
+			// Filter to allow users to modify the email subject.
+			$subject = apply_filters( 'tribe-ext-organizer-notifications-subject', 'Your event %1$s has new attendee(s) - %2$s' );
+
+			// Return the subject with event and site names injected.
+			return sprintf( __( $subject, 'tribe-extension' ), get_the_title( $post_id ), get_bloginfo( 'name' ) );
```
Comment (Camwyn) [inline tribe-ext-organizer-notifications.php]: Not your code but a generic text domain like this is just asking for conflicts down the road?

Comment (vicskf) [reply tribe-ext-organizer-notifications.php]: Thanks @Camwyn ! I fixed this and also on another place.

**Comment by vicskf** [comment]
@Camwyn Sorry I requested the code review before. Now it's good to review

**Review by zbtirrell — APPROVED** [review]

**Comment by vicskf** [comment]
> > Updated the get_recipient function to fetch emails from all organizers linked to the post
> > Ticket: EXT-212
> 
> I have installed this plug in to Event Tickets Pro, but it is still only emailing the first organiser

Hi @bobr666 ! Thanks for letting us know about that. We have just released version 1.0.1 that includes support for multiple organizers https://github.com/mt-support/tribe-ext-organizer-notifications/releases/tag/1.0.1

**Pull request merged**